src/components/RwRating/RwRating.vue

```diff
@@ -0,0 +1,118 @@
+<script setup lang="ts">
+import { useVModel } from '@vueuse/core';
+import { computed, ref } from 'vue';
+import RwIcon from '../RwIcon/RwIcon.vue';
+// import { RwRatingEvents, RwRatingProps } from './types';
+// interface Props extends Omit<RwRatingProps, keyof RwRatingEvents> {}
+
+interface RwRatingEvents {
+  ['onUpdate:modelValue']: () => void;
+}
+interface RwRatingProps extends RwRatingEvents {
+  modelValue: number;
+  count?: number;
+}
+
+const props = withDefaults(defineProps<RwRatingProps>(), {
+  count: 5,
+});
+
+interface Emits {
+  (e: 'update:modelValue', value: boolean): void;
+}
+const emit = defineEmits<Emits>();
+
+const value = useVModel(props, 'modelValue', emit);
+
+const currentIndex = ref(value.value);
+
+function getCurrentIconName(countNumber: number) {
+  if (countNumber <= Math.floor(currentIndex.value)) {
+    return 'fullSlot';
+  }
+
+  if (countNumber === Math.ceil(currentIndex.value)) {
+    return 'partSlot';
+  }
+
+  return 'emptySlot';
+}
+
+const icons = computed(() => {
+  const iconList: string[] = [];
+  for (let i = 0; i < props.count; i += 1) {
+    iconList.push(getCurrentIconName(i + 1));
+  }
```

Comment:
for (let i = 1; i < props.count + 1; i += 1) {
    iconList.push(getCurrentIconName(i));
  }